File: csharp/series/Series.cs
using System;

public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength > numbers.Length || sliceLength <= 0)
            throw new ArgumentException();

        var numberOfSlices = numbers.Length - sliceLength + 1;

        var output = new string[numberOfSlices];

        for (var i = 0; i < numberOfSlices; ++i)
            output[i] = numbers.Substring(i, sliceLength);

        return output;
    }
}

File: javascript/series/series.js
export class Series {
    constructor(str) {
        this.digits = str.split("").map(n => parseInt(n));
    }

    slices(n) {
        if (n > this.digits.length)
            throw "Slice size is too big.";

        let output = [];
        let numberOfSlices = this.digits.length - n + 1;

        for (let i = 0; i < numberOfSlices; ++i)
            output.push(this.digits.slice(i, i + n));

        return output;
    }
}
